terminology/src/test/java/au/csiro/pathling/terminology/TerminologyAuthenticationTest.java

```diff
@@ -0,0 +1,275 @@
+/*
```

Comment:
I have a feeling that with some refactoring most of this could be tested in unit tests that do not require WireMock. 
Essentially we have the `AuthInterceptor` that needs access to an `AccessContextProvider`.
The `AccessContextProvider` can come in two flavours 
 - HttpAccessContextProvider that handles the actual logic of making the HTTP request to the endpoint.
 - SharedCachedAccessContextProvider that handles the expiration and sharing, and if needed makes the call to the `HttpAccessContextProvider`
 
With such (or similar) design I think most of the logic can be tested with Mockito mocks.

With WireMock maybe only needed for an end-to-end integration test and possibly to test `HttpAccessContextProvider`

Also I did notice that that actual WireMock integration test for TerminolgyService (`au.csiro.pathling.test.integration.TerminologyServiceIntegrationTest`) is still in `fhir-server` module.  Maybe we could put both test in the same module (not sure which one).